chore(tracking): Release Notes for SDP 25.7.0 (#757)
stackabletech/documentation: modules/ROOT/partials/release-notes/release-25.7.adoc

Refer to the xref:operators:supported_versions.adoc[supported versions] documentation for a complete overview including LTS versions or deprecations.

===== New versions

The following new product versions are now supported:

* ...

===== Deprecated versions

The following product versions are deprecated and will be removed in a later release:

* ...

===== Removed versions

The following product versions are no longer supported (although images for released product versions remain available https://oci.stackable.tech/[here,window=_blank]. Information on how to browse the registry can be found xref:contributor:project-overview.adoc#docker-images[here,window=_blank].):

* ...

==== stackablectl

==== Supported Kubernetes versions

This release supports the following Kubernetes versions:

* `1.XX`

These Kubernetes versions are no longer supported:

* `1.XX`

==== Supported OpenShift versions

This release is available in the RedHat Certified Operator Catalog for the following OpenShift versions:

* `4.XX`

These OpenShift versions are no longer supported:

* `4.XX`

==== Breaking changes

> Of the changes mentioned above, the following are breaking (or could lead to breaking behaviour), and you will need to adapt your existing CRDs accordingly:

Reviewer: I'll try to refine them a bit here:

- https://github.com/stackabletech/issues/issues/692 (some release notes a here, others follow...)
  - Airflow + Superset: https://github.com/stackabletech/issues/issues/692#issuecomment-2854266967
  - Spark: https://github.com/stackabletech/issues/issues/692#issuecomment-2847025199
  - Trino: https://github.com/stackabletech/issues/issues/692#issuecomment-3001089387
  - ZooKeeper: https://github.com/stackabletech/issues/issues/692#issuecomment-3034672506
  - Druid: no release note written @labrenbe
  - NiFi: No release-note written @labrenbe 
  - HBase: No release-note written @adwk67 
  - Hive: No release-note written @Maleware 
- https://github.com/stackabletech/operator-rs/pull/1054
- https://github.com/stackabletech/kafka-operator/pull/868
- https://github.com/stackabletech/docker-images/pull/1124

> Of the changes mentioned above, the following are breaking (or could lead to breaking behaviour), and you will need to adapt your existing CRDs accordingly:

Reviewer: That seems to be everything that was tagged as breaking.

> Of the changes mentioned above, the following are breaking (or could lead to breaking behaviour), and you will need to adapt your existing CRDs accordingly:

Author: Thank @NickLarsenNZ, the summary is highly appreciated!

> Of the changes mentioned above, the following are breaking (or could lead to breaking behaviour), and you will need to adapt your existing CRDs accordingly:

Author: Added in d9771697.

kubectl replace -f https://raw.githubusercontent.com/stackabletech/commons-operator/25.7.0/deploy/helm/commons-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/druid-operator/25.7.0/deploy/helm/druid-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/hbase-operator/25.7.0/deploy/helm/hbase-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/hdfs-operator/25.7.0/deploy/helm/hdfs-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/hive-operator/25.7.0/deploy/helm/hive-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/kafka-operator/25.7.0/deploy/helm/kafka-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/listener-operator/25.7.0/deploy/helm/listener-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/nifi-operator/25.7.0/deploy/helm/nifi-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/opa-operator/25.7.0/deploy/helm/opa-operator/crds/crds.yaml

> kubectl replace -f https://raw.githubusercontent.com/stackabletech/secret-operator/25.7.0/deploy/helm/secret-operator/crds/crds.yaml

Reviewer 2: Watch out, I think we need something a bit more advanced here.
We should definitely check this commands :)

Author: Changed in d3c9b6ac.

Reviewer 2: Tested this using `stackablectl re in 25.3 `
Afterwards we need to following script:
```bash
kubectl replace -f https://raw.githubusercontent.com/stackabletech/airflow-operator/25.7.0/deploy/helm/airflow-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/commons-operator/25.7.0/deploy/helm/commons-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/druid-operator/25.7.0/deploy/helm/druid-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/hbase-operator/25.7.0/deploy/helm/hbase-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/hdfs-operator/25.7.0/deploy/helm/hdfs-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/hive-operator/25.7.0/deploy/helm/hive-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/kafka-operator/25.7.0/deploy/helm/kafka-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/listener-operator/25.7.0/deploy/helm/listener-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/nifi-operator/25.7.0/deploy/helm/nifi-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/opa-operator/25.7.0/deploy/helm/opa-operator/crds/crds.yaml
# We have a new CRD: truststores.secrets.stackable.tech
kubectl replace -f https://raw.githubusercontent.com/stackabletech/secret-operator/25.7.0/deploy/helm/secret-operator/crds/crds.yaml || true
# We have a new CRD: sparkconnectservers.spark.stackable.tech
kubectl create -f https://raw.githubusercontent.com/stackabletech/secret-operator/25.7.0/deploy/helm/secret-operator/crds/crds.yaml || true
kubectl replace -f https://raw.githubusercontent.com/stackabletech/spark-k8s-operator/25.7.0/deploy/helm/spark-k8s-operator/crds/crds.yaml || true
kubectl create -f https://raw.githubusercontent.com/stackabletech/spark-k8s-operator/25.7.0/deploy/helm/spark-k8s-operator/crds/crds.yaml || true
kubectl replace -f https://raw.githubusercontent.com/stackabletech/superset-operator/25.7.0/deploy/helm/superset-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/trino-operator/25.7.0/deploy/helm/trino-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/zookeeper-operator/25.7.0/deploy/helm/zookeeper-operator/crds/crds.yaml
```
Can you please update it?

helm install --wait hdfs-operator oci://oci.stackable.tech/sdp-charts/hdfs-operator --version 25.7.0
helm install --wait hive-operator oci://oci.stackable.tech/sdp-charts/hive-operator --version 25.7.0
helm install --wait kafka-operator oci://oci.stackable.tech/sdp-charts/kafka-operator --version 25.7.0
helm install --wait listener-operator oci://oci.stackable.tech/sdp-charts/listener-operator --version 25.7.0
helm install --wait nifi-operator oci://oci.stackable.tech/sdp-charts/nifi-operator --version 25.7.0
helm install --wait opa-operator oci://oci.stackable.tech/sdp-charts/opa-operator --version 25.7.0
helm install --wait secret-operator oci://oci.stackable.tech/sdp-charts/secret-operator --version 25.7.0
helm install --wait spark-k8s-operator oci://oci.stackable.tech/sdp-charts/spark-k8s-operator --version 25.7.0
helm install --wait superset-operator oci://oci.stackable.tech/sdp-charts/superset-operator --version 25.7.0
helm install --wait trino-operator oci://oci.stackable.tech/sdp-charts/trino-operator --version 25.7.0
helm install --wait zookeeper-operator oci://oci.stackable.tech/sdp-charts/zookeeper-operator --version 25.7.0
----

> ==== Known issues

Reviewer 3: I've added my suggestion. Feel free to take or edit it as you need

* The operator now emits a warning for deprecated sensitive property algorithms in NiFi version 1.x.x and errors out for version 2.x.x if an unsupported sensitive properties algorithm is used.
  See the official https://nifi.apache.org/docs/nifi-docs/html/administration-guide.html#property-encryption-algorithms[NiFi documentation] and https://github.com/stackabletech/nifi-operator/issues/791[nifi-operator#791].
* Re-add Iceberg support to NiFi 2.4.0, which was removed upstream starting with version 2.0.0.
  See xref:nifi:usage_guide/writing-to-iceberg-tables.adoc[NiFi Iceberg documentation] and https://github.com/stackabletech/nifi-operator/issues/738[nifi-operator#738].

===== Apache ZooKeeper

The image for Apache ZooKeeper backports various 3.9.4 changes into the supported 3.9.3 version.
See https://github.com/stackabletech/docker-images/issues/1143[docker-images#1143].

==== Platform deprecations

==== Product versions

As with previous SDP releases, many product images have been updated to their latest versions.

Refer to the xref:operators:supported_versions.adoc[supported versions] documentation for a complete overview including LTS versions or deprecations.

> ===== New versions

Reviewer: Double check these against the spreadsheet, at least Hive 3.1.3 and 4.0.1 deprecations are missing...

Author: As far as I can see, these version were already deprecated in 25.3.0. The only non-deprecated and non-experimental version is 4.0.0 which was introduced as the LTS in 25.3.0.

But I can list 3.1.3 again, mentioning it remains deprecated since 25.3.0.

Reviewer: I don't think it needs mentioning if it was done in a previous release.

The following new product versions are now supported:

* Apache Airflow: https://github.com/stackabletech/docker-images/issues/1074[2.10.5], https://github.com/stackabletech/docker-images/issues/1074[3.0.1 (experimental)]
* Apache Druid: https://github.com/stackabletech/docker-images/issues/1075[33.0.0]
* Apache HBase: https://github.com/stackabletech/docker-images/issues/1076[2.6.2]
* Apache Kafka: https://github.com/stackabletech/docker-images/issues/1078[4.0.0 (experimental)^1^]
* Apache NiFi: https://github.com/stackabletech/docker-images/issues/1079[2.4.0]
* Apache Superset: https://github.com/stackabletech/docker-images/issues/1081[4.1.2]
* Open Policy Agent: https://github.com/stackabletech/docker-images/issues/1084[1.4.2]
* Trino: https://github.com/stackabletech/docker-images/issues/1082[476]

===== Deprecated versions

The following product versions are deprecated and will be removed in a later release:

* Apache Airflow: https://github.com/stackabletech/docker-images/issues/1074[2.10.4]
* Apache Druid: https://github.com/stackabletech/docker-images/issues/1075[31.0.1]
* Apache Hbase: https://github.com/stackabletech/docker-images/issues/1076[2.6.1]

> * Apache Superset: https://github.com/stackabletech/docker-images/issues/1081[4.2.1]

Author: See https://github.com/stackabletech/documentation/pull/757#discussion_r2215273731.

// Here are the headings you can use for the next release. Saves time checking indentation levels.
// Take a look at release 24.11 to see how to structure patch releases.

== Release 25.7

=== 25.7.0

> Released on YYYY-MM-DD.

Reviewer: ```suggestion
Released on 2025-07-23.
```

> (Optional description / introduction)

Reviewer: Todo?

Author: Yeah, there seems to be no particular theme other than listener-related changes. Also, I'm not the most creative one to come up with a description/summary/introduction.

[NOTE]
====
The SDP 25.7.0 release is the first release which  will **only** be published on `oci.stackable.tech`.
====

[NOTE]
====

> In previous releases, releases of the `stackablectl` CLI tool were synced with platform releases.

Reviewer: ```suggestion
Previously, releases of the `stackablectl` CLI tool were synced with platform releases.
```

The tool itself was never tied to a particular version of SDP.
As such, it was decided to version `stackablectl` independently.
// Starting with this release, the release notes for `stackablectl` are located on a separate xref:management:stackablectl/release-notes.adoc[page].

> Starting with this release, the release notes for `stackablectl` are located on a separate page.

Reviewer: We should probably link the page here so it is directly accessible, eg:

```suggestion
Starting with this release, the release notes for `stackablectl` are located on a xref::blah[separate page].
```

====

==== New platform features

===== General

====== Security

* Users (and services) can now use the new TrustStore resource to request the trust root certificates associated with a SecretClass.
  See the xref:secret-operator:truststore.adoc[TrustStore documentation] and https://github.com/stackabletech/secret-operator/issues/410[secret-operator#410].
* Add support for format-specific annotations to override secret file names.
  Names can be customized using secret volume annotations which are listed xref:secret-operator:volume.adoc[in our documentation].
  See https://github.com/stackabletech/secret-operator/pull/572[secret-operator#572].

====== Automatic cluster domain detection

The default Kubernetes cluster domain name is now fetched from the kubelet API unless explicitly configured.

> Operators are now required to have the RBAC permission to get nodes/proxy in the apiGroup `""`.

Reviewer: I feel like the empty apiGroup is redundant. How about:

```suggestion
Operators are now required to have the RBAC permission to get `nodes/proxy`.
```

The helm-charts takes care of this.

The CLI argument `--kubernetes-node-name` or env variable `KUBERNETES_NODE_NAME` needs to be set.
The helm-charts takes care of this.

See https://github.com/stackabletech/issues/issues/662[issues#662].

====== Observability

> * All operators support sending OpenTelemetry logs and traces.

Reviewer: ```suggestion
* All operators support sending OpenTelemetry logs and traces via OTLP.
```

===== Apache Hive

Hive 3 has been deprecated since SDP 25.3.0 and was marked as End of Life on 2024-10-08 by the upstream project.
We might remove it in the upcoming release, the final decision will be based on metastore client compatibility and customer feedback.

===== Apache Spark

This release enables the built in Prometheus servlet for metric scraping.
The existing JMX exporter is still enabled but it is deprecated and it will be removed in a future release.
See https://github.com/stackabletech/spark-k8s-operator/pull/584[spark-k8s-operator#584].

==== Platform removals

===== General

* Remove CSI registration path migration job (init-container) from secret-operator DaemonSet.
  See https://github.com/stackabletech/secret-operator/pull/610[secret-operator#610].
* This release removes support for previously deprecated product service accounts.
  Deprecated service accounts named `<product_name>-serviceaccount` are ignored.
  Instead, product workloads use service accounts named `<resource_name>-serviceaccount` where `resource_name` is the stacklet (or cluster) name as defined in the cluster resource.

>   See https://github.com/stackabletech/operator-rs/pull/1060[operator-rs#1060].

Reviewer: Maybe we need a _call to action_ for deleting the orphaned/unused SAs.

Author: Added in f64eb396.

* *Breaking:*
  The `lastUpdatedTime` was removed from the `ClusterStatus` of each operator.
  Users previously relying on this value should migrate to `lastTransitionTime`.
  See https://github.com/stackabletech/operator-rs/issues/1032[operator-rs#1032] and https://github.com/stackabletech/operator-rs/pull/1054[operator-rs#1054].

===== Container images

*Breaking:*
The `oci.stackable.tech/sdp/kcat` image has been discontinued starting with SDP 25.7.0.
Use `oci.stackable.tech/sdp/kafka-testing-tools` instead, which includes kcat.
Existing `oci.stackable.tech/sdp/kcat` images for previous SDP releases remain available.
See https://github.com/stackabletech/docker-images/pull/1124[docker-images#1124].

===== Apache HBase

> Starting with this release, support for HBase 2.4.x has been removed.

Reviewer: ```suggestion
Starting with this release, support for HBase 2.4 has been removed.
```

* `4.18`
* `4.17`
* `4.16`

These OpenShift versions are no longer supported:

* `4.15`
* `4.14`

==== Upgrade from 25.3

===== Using stackablectl

====== Upgrade with a single command

Starting with stackablectl Release 1.0.0 the multiple consecutive commands described below can be shortened to just one command, which executes exactly those steps on its own.

[source,console]
----
$ stackablectl release upgrade 25.7
----

====== Upgrade with multiple consecutive commands

Uninstall the `25.3` release

[source,console]
----
$ stackablectl release uninstall 25.3

Uninstalled release '25.3'

Use "stackablectl release list" to list available releases.
# ...
----

Afterwards you will need to upgrade the CustomResourceDefinitions (CRDs) installed by the Stackable Platform.
The reason for this is that helm will uninstall the operators but not the CRDs.
This can be done using `kubectl replace`.

[IMPORTANT]
====
The Stackable secret operator CRD needs to be applied in addition to being replaced, because it introduces the new TrustStore custom resource.
The commands below are adjusted accordingly.

> This issue is also mentioned in the known issues section below.

Reviewer: We should link this

Author: Done in 760e9f1c.

helm install --wait hdfs-operator oci://oci.stackable.tech/sdp-charts/hdfs-operator --version 25.7.0
helm install --wait hive-operator oci://oci.stackable.tech/sdp-charts/hive-operator --version 25.7.0
helm install --wait kafka-operator oci://oci.stackable.tech/sdp-charts/kafka-operator --version 25.7.0
helm install --wait listener-operator oci://oci.stackable.tech/sdp-charts/listener-operator --version 25.7.0
helm install --wait nifi-operator oci://oci.stackable.tech/sdp-charts/nifi-operator --version 25.7.0
helm install --wait opa-operator oci://oci.stackable.tech/sdp-charts/opa-operator --version 25.7.0
helm install --wait secret-operator oci://oci.stackable.tech/sdp-charts/secret-operator --version 25.7.0
helm install --wait spark-k8s-operator oci://oci.stackable.tech/sdp-charts/spark-k8s-operator --version 25.7.0
helm install --wait superset-operator oci://oci.stackable.tech/sdp-charts/superset-operator --version 25.7.0
helm install --wait trino-operator oci://oci.stackable.tech/sdp-charts/trino-operator --version 25.7.0
helm install --wait zookeeper-operator oci://oci.stackable.tech/sdp-charts/zookeeper-operator --version 25.7.0
----

[#known-issues-25_7_0]

> ==== Known issues

Reviewer: From @adwk67:

> I've added a release note, basically to the effect that "everything listener-related is breaking", but if we can confirm that re-deploying the stacklet is the way to go, this would add helpful detail.
